src/mesh.cpp

```diff
@@ -186,15 +186,13 @@ void shadeMeshFaces(scene::IMesh *mesh)
 		for (u32 i = 0; i < vertex_count; i++) {
 			video::S3DVertex *vertex = (video::S3DVertex *)(vertices + i * stride);
 			video::SColor &vc = vertex->Color;
+			// Many special drawtypes have normals set to 0,0,0 and this
+			// must result in maximum brightness (no face shadng).
 			if (vertex->Normal.Y < -0.5) {
 				applyFacesShading (vc, 0.447213);
-			} else if (vertex->Normal.Z > 0.5) {
+			} else if (vertex->Normal.X > 0.5 || vertex->Normal.X < -0.5) {
```

Comment:
The PR comment said that floating point literals were converted to float. 0.5 is a double.